vaadin-spring/src/main/java/com/vaadin/flow/spring/LaunchUtil.java

```diff
@@ -0,0 +1,137 @@
+package com.vaadin.flow.spring;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletContext;
+
+import com.vaadin.flow.server.VaadinContext;
+import com.vaadin.flow.server.VaadinServletContext;
+import com.vaadin.flow.server.frontend.FrontendTools;
+import com.vaadin.flow.server.frontend.FrontendUtils;
+import com.vaadin.flow.server.startup.ApplicationConfiguration;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.web.context.support.GenericWebApplicationContext;
+
+/**
+ * Utilities for launching a browser when running in development mode.
+ */
+public class LaunchUtil {
+
+    private static final String LAUNCH_TRACKER = "LaunchUtil.hasLaunched";
+    private static final String LAUNCHED_VALUE = "yes";
+    private static File bundleTempFile;
+
+    private LaunchUtil() {
+        // Static methods only
+    }
+
+    /**
+     * Launch the default browser and open the given application base URL if
+     * running in development mode.
+     *
+     * Does nothing if the application is running in production mode.
+     */
+    public static void launchBrowserInDevelopmentMode(
+            ConfigurableApplicationContext app) {
+        if (isLaunched()) {
+            // Only launch browser on startup, not on reload
+            return;
+        }
+        if (!LaunchUtil.isProductionMode(app)) {
+            String location = "http://localhost:"
+                    + app.getEnvironment().getProperty("local.server.port")
+                    + "/";
```

Comment:
Should this open always on "/" or would it be better to add the context path as well? ServletContext should be available

Reply:
Good catch. This was never supported by the helper library but now it supports both context path and Vaadin mapping